### PR TITLE
Jdbc/spring boot sample

### PR DESCRIPTION
**Pull request by nsundara**
Hi Kuassi, 

Can you please review the changes? 

Regards,
Nirmala

### CODE REVIEW

File: java/jdbc/SpringBootSample/src/main/java/com/oracle/springapp/DataSourceConfig.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Primary;
+
+import oracle.ucp.jdbc.PoolDataSource;
+import oracle.ucp.jdbc.PoolDataSourceFactory;
+
+/**
+ * This will serve as your primary data source
+ * If you do not want to use this as primary datasource remove the @Primary
+ * and if you want to use OracleUniversalConnectionPool you can add @Qualified
```
Comment (Kuassim) [inline java/jdbc/SpringBootSample/src/main/java/com/oracle/springapp/DataSourceConfig.java:18]: As UCP is used, shouldn't we say "If you don't want to use UCP ...remove @Qualified"?

File: java/jdbc/SpringBootSample/src/main/java/com/oracle/springapp/dao/EmployeeDAO.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package com.oracle.springapp.dao;
+
+import java.util.List;
+
+import com.oracle.springapp.model.Employee;
+
+/**
+ * Simple DAO interface for EMP table.
```
Comment (Kuassim) [inline java/jdbc/SpringBootSample/src/main/java/com/oracle/springapp/dao/EmployeeDAO.java:8]: This is not for the EMP table

File: java/jdbc/SpringBootSample/src/main/java/com/oracle/springapp/dao/AllTablesDAO.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package com.oracle.springapp.dao;
+
+import java.util.List;
+
+import com.oracle.springapp.model.AllTables;
+
+/**
+ * Simple DAO interface for EMP table.
```
Comment (Kuassim) [inline java/jdbc/SpringBootSample/src/main/java/com/oracle/springapp/dao/AllTablesDAO.java:8]: Is this for AllTables or EMP table?

File: java/jdbc/SpringBootSample/src/main/java/com/oracle/springapp/dao/impl/EmployeeDAOImpl.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+	
+	@Override
+	public void insertEmployee(Employee employee) {
+		String sql = "Select max(empno) from emp"; 
+	    int[] emp_no = new int[1];
+	    //getJdbcTemplate().query(sql, (rs, rowNum) -> emp_no[rowNum] = rs.getInt(1));
+	    getJdbcTemplate().query(sql, (resultSet, rowNumber) -> emp_no[rowNumber] = resultSet.getInt(1));
+
+	    
+	    System.out.println("Max emploee number is: " + emp_no[0]);
```
Comment (Kuassim) [inline java/jdbc/SpringBootSample/src/main/java/com/oracle/springapp/dao/impl/EmployeeDAOImpl.java:63]: Typo employee

File: java/jdbc/SpringBootSample/src/main/resources/application.properties [hunk 5]
```diff
@@ -0,0 +1,6 @@
+spring.datasource.url=jdbc:oracle:thin:@ofunctions_medium?TNS_ADMIN=/Users/nbsundar/ATPTesting/wallet_OFUNCTIONS
+spring.datasource.username=testuser
+spring.datasource.password=WE10come12##
```
Comment (Kuassim) [inline java/jdbc/SpringBootSample/src/main/resources/application.properties:3]: Do you want to leave the actual username and password?

File: java/jdbc/SpringBootSample/src/main/java/com/oracle/springapp/dao/impl/AllTablesDAOImpl.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+	@PostConstruct
+	public void initialize() {
+		setDataSource(dataSource);
+		System.out.println("Datasource used: " + dataSource);
+	}
+
+	@Override
+	public List<AllTables> getTableNames() {
+		final String sql = "SELECT owner, table_name, status, num_rows FROM all_tables where rownum < 20";
```
Comment (Kuassim) [inline java/jdbc/SpringBootSample/src/main/java/com/oracle/springapp/dao/impl/AllTablesDAOImpl.java:32]: Please use a PreparedStatement with a bind variable; this is the best practice we have been preaching.

File: java/jdbc/SpringBootSample/src/main/java/com/oracle/springapp/dao/EmployeeDAO.java [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+
+import com.oracle.springapp.model.Employee;
+
+/**
+ * Simple DAO interface for EMP table.
+ *
+ */
+public interface EmployeeDAO {
+	public List<Employee> getAllEmployees();
+
```
Comment (Kuassim) [inline java/jdbc/SpringBootSample/src/main/java/com/oracle/springapp/dao/EmployeeDAO.java:13]: Perhaps add a getEmployee( Empno empno) which prints only  a specific employee?

**Comment by Kuassim** [comment]
Nirmala,

I've made some comments and suggestions.

Let me know if you cannot see these.

Thanks, Kuassi

File: java/jdbc/SpringBootSample/src/main/java/com/oracle/springapp/DataSourceConfig.java (hunk 1 above)
Comment (nsundara) [reply java/jdbc/SpringBootSample/src/main/java/com/oracle/springapp/DataSourceConfig.java:18]: If the customers don't want to use UCP then, they don't even need this DataSourceConfig.java. Spring deduces that it is Oracle JDBC connection based on the URL read from the application.properties file.

File: java/jdbc/SpringBootSample/src/main/java/com/oracle/springapp/dao/AllTablesDAO.java (hunk 3 above)
Comment (nsundara) [reply java/jdbc/SpringBootSample/src/main/java/com/oracle/springapp/dao/AllTablesDAO.java:8]: This is for AllTables. I have a separate one for EmployeeDAO.java

Comment (Kuassim) [reply java/jdbc/SpringBootSample/src/main/java/com/oracle/springapp/dao/AllTablesDAO.java:8]: Exactly my point; the comment says "Simple DAO interface for EMP table"; this is a copy/paste typo.

File: java/jdbc/SpringBootSample/src/main/java/com/oracle/springapp/DataSourceConfig.java (hunk 1 above)
Comment (Kuassim) [reply java/jdbc/SpringBootSample/src/main/java/com/oracle/springapp/DataSourceConfig.java:18]: If so, why add the comment?
" ... and if you want to use OracleUniversalConnectionPool you can add @Qualified"
Is there a risk of conflict?

**Pull request closed without merging**